ui/appui-react/src/appui-react/syncui/SyncUiEventDispatcher.ts

```diff
@@ -66,91 +64,182 @@ export enum SyncUiEventId {
   ViewedModelsChanged = "viewedmodelschanged",
 }
 
+/**
+ * UiSync Event arguments. Contains a set of lower case event Ids.
+ * @public
+ */
+export interface UiSyncEventArgs {
+  eventIds: Set<string>;
+}
+
+/** UiSync Event class.
+ * @public
+ */
+export class UiSyncEvent extends BeUiEvent<UiSyncEventArgs> {}
+
 /** This class is used to send eventIds to interested UI components so the component can determine if it needs
  * to refresh its display by calling setState on itself.
  * @public
  */
 export class SyncUiEventDispatcher {
-  private static _uiEventDispatcher = new UiEventDispatcher();
   private static _unregisterFuncs = new Array<() => void>();
   private static _connectionUnregisterFuncs = new Array<() => void>();
   private static _iModelConnection?: IModelConnection;
+  private static _syncEventTimerId: number | undefined;
```

Comment:
This is going backwards though, we want to move toward less statics, you can move the code to something like `InternalSyncUiEventDispatcher.tsx` and likely leave it as it is currently in `appui-abstract`, we will want to move more of the logic to that class and remove the specifics from the static class rather that the other way around.

Reply:
I will take a look, thanks.